pkg/kubelet/kubelet.go

```diff
@@ -2426,28 +2433,40 @@ func (kl *Kubelet) cleanUpContainersInPod(podID types.UID, exitedContainerID str
 }
 
 // fastStatusUpdateOnce starts a loop that checks the internal node indexer cache for when a CIDR
-// is applied  and tries to update pod CIDR immediately. After pod CIDR is updated it fires off
-// a runtime update and a node status update. Function returns after one successful node status update.
+// is applied and tries to update pod CIDR immediately. Given the pod CIDR is applied, it keeps trying
+// to update pod CIDR then container runtime status and node status, until node status becomes ready.
+// Function returns after one successful node status update that gets the node to the ready status.
 // Function is executed only during Kubelet start which improves latency to ready node by updating
 // pod CIDR, runtime status and node statuses ASAP.
 func (kl *Kubelet) fastStatusUpdateOnce() {
+	defer func() {
+		kl.containerRuntimeReadyExpected = true
+	}()
+
+	start := time.Now()
```

Comment:
```suggestion
	start := kl.clock.Now()
```

we can inject one fake for testing